src/main/java/nextstep/auth/authentication/UsernamePasswordAuthenticationFilter.java

```diff
@@ -15,7 +18,28 @@ public UsernamePasswordAuthenticationFilter(LoginMemberService loginMemberServic
 
     @Override
     public boolean preHandle(HttpServletRequest request, HttpServletResponse response, Object handler) {
-        // TODO: 구현하세요.
-        return true;
+        try {
+            String principal = request.getParameter("username");
+            String credentials = request.getParameter("password");
+
+            AuthenticationToken token = new AuthenticationToken(principal, credentials);
```

Comment:
```suggestion
           Authentication authentication = getAuthentication(request)
```
이런식으로 로직을 분리해서 가독성을 높힐 수 있지 않을까요